docsrc/sample.rst

```diff
@@ -70,28 +71,52 @@ The :ref:`class_cmdstanmodel` class method  ``sample`` returns a ``CmdStanMCMC``
+In short, the sampler is configuration specifies iterations; the
+results specify draws.
+
+Summarizing and diagnosing the fitted model:
 
 - ``summary()`` - Run CmdStan's `stansummary <https://mc-stan.org/docs/cmdstan-guide/stansummary.html>`__ utility on the sample.
 - ``diagnose()`` - Run CmdStan's `diagnose <https://mc-stan.org/docs/cmdstan-guide/diagnose.html>`__ utility on the sample.
 - ``sampler_diagnostics()`` - Returns the sampler parameters as a map from sampler parameter names to a numpy.ndarray of dimensions draws X chains X 1.
 
-Methods for managing the sample:
+Information about the size and shape of the sample: 
 
-- ``save_csvfiles(dir_name)`` - Move output csvfiles to specified directory.
 - ``chains`` - Number of chains
-- ``num_draws`` - Number of post-warmup draws (i.e., sampling iterations)
-- ``num_warmup_draws`` - Number of warmup draws.
-- ``metric`` - Per chain metric by the HMC sampler.
-- ``stepsize`` - Per chain stepszie used by the HMC sampler.
-- ``sample`` - A 3-D numpy.ndarray which contains all post-warmup draws across all chains arranged as (draws, chains, columns).
-- ``warmup`` - A 3-D numpy.ndarray which contains all warmup draws across all chains arranged as (draws, chains, columns).
+- ``num_draws`` - Number of draws per chain, i.e., thinned iterations.
+- ``num_draws_sampling`` - Number of sampling (post-warmup) draws per chain, i.e., sampling iterations, thinned.  By default, only the post-warmup draws are reported, so that ``num_draws`` == ``num_draws_sampling``.
+- ``num_draws_warmup`` - Number of warmup draws per chain, i.e., thinned warmup iterations.
+
+- ``metric_type`` - Metric type used for adaptation, either ``diag_e``
+  or ``dense_e``, or ``None``, if the Stan program doesn't have any parameters.
+- ``num_params`` - Total number of parameters in the model; the sum of all scalar parameter variables and all elements of all container parameter variables
+
+- ``column_names`` - Column labels for one draw from the sampler.
+- ``sampler_vars_cols`` - Maps the sampler parameter names to output column indices.
+- ``stan_vars_cols`` - Maps the Stan progran variable names to output column indices.
+- ``stan_vars_dims`` - Maps the Stan progran variable names to dimensions; scalar variables have zero dimensions.
 
-Methods for downstream analysis are:
+Contents of the the sample: 
 
+- ``metric`` - Per-chain metric values, either a vector or matrix. 
+- ``step_size`` - Per-chain stepsize.
+  
+- ``draws`` - A numpy.ndarray which contains all across all chains arranged as (raws, chains, columns).
 - ``stan_variable(var_name)`` - Returns a numpy.ndarray which contains the set of draws in the sample for the named Stan program variable.
 - ``stan_variables()`` - Return dictionary of all Stan program variables.
 
+
+Methods for saving the CmdStan output files:
+
+- ``save_csvfiles(dir_name)`` - Move output Stan CSV files to specified directory. 
```

Comment:
Should there be an underscore here in the csvfiles name? See:

```
>>> fit.save_csvfiles
<bound method CmdStanMCMC.save_csvfiles of CmdStanMCMC: model=test chains=4['method=sample', 'algorithm=hmc', 'adapt', 'engaged=1']
 csv_files:
        /tmp/tmpwxcqwf_l/test-202102090856-1-cssacms9.csv
        /tmp/tmpwxcqwf_l/test-202102090856-2-w1hcupqr.csv
        /tmp/tmpwxcqwf_l/test-202102090856-3-3uggqmdv.csv
        /tmp/tmpwxcqwf_l/test-202102090856-4-gc51buo8.csv
 output_files:
        /tmp/tmpwxcqwf_l/test-202102090856-1-cssacms9-stdout.txt
        /tmp/tmpwxcqwf_l/test-202102090856-2-w1hcupqr-stdout.txt
        /tmp/tmpwxcqwf_l/test-202102090856-3-3uggqmdv-stdout.txt
        /tmp/tmpwxcqwf_l/test-202102090856-4-gc51buo8-stdout.txt>
```

Reply:
probably - but that was what was decided on with a lot of back and forth, so reluctant to change this now.